sqlmesh/core/console.py

```diff
@@ -493,14 +512,56 @@ def update_migration_progress(self, num_tasks: int) -> None:
         if self.migration_progress is not None and self.migration_task is not None:
             self.migration_progress.update(self.migration_task, refresh=True, advance=num_tasks)
 
-    def stop_migration_progress(self, success: bool = True) -> None:
+    def log_migration_status(self, success: bool = True) -> None:
+        """Log the migration status."""
+        if self.migration_progress is not None:
+            self.migration_progress = None
+            if success:
+                self.log_success("The migration has been completed successfully")
+
+    def stop_snapshot_migration_progress(self, success: bool = True) -> None:
         """Stop the migration progress."""
         self.migration_task = None
         if self.migration_progress is not None:
             self.migration_progress.stop()
-            self.migration_progress = None
```

Comment:
Why did this need to be changed?

Reply:
I guess because we set it to `None` in `stop_env_migration_progress`? Should we be using two separate attributes, i.e. `self.snapshot_migration_progress` and `self.env_migration_progress`?

Reply:
This was to accommodate logging messages for when environment and snapshot migrations have completed and then a final message for the migration. To cover cases for example when snapshot migrations might happen but not environment.